src/lib/container/linked_cell.h

```diff
@@ -0,0 +1,398 @@
+    });
+    SPDLOG_TRACE("fixed positions");
+  }
+
+ private:
+  /**
+   * Inserts a reference to a particle in a arena into it's cell
+   * */
+  auto insert_into_cell(arena<Particle>::entry &particle) {
+
+    auto idx = index.position_to_index(particle.data.position);
+    if (idx < cells.size()) {
+      cells[idx].insert(particle);
+    } else {
+      spdlog::warn("a particle has positions that is out of bounds {} {} {}", particle.data.position[0], particle.data.position[1], particle.data.position[2]);
+    }
+  }
+
+  container::arena<Particle> arena;
+  std::vector<cell> cells;
+  std::array<BoundaryCondition, 6> bc;
+
+ public:
+  I index;
+  double sigma;
+  double cutoff;
+};
+
+/**
+ * a function to apply all up to 6 boundary conditions to a all boundary cell of a linked container
+ * */
+template<index::Index I>
+static void calculate_boundary_condition(linked_cell<I> &lc,
+                                         std::function<void(std::tuple<Particle &, Particle &>)> const &force_calculation
+
+) {
+
+  std::ranges::for_each(lc.boundary(), [&lc, &force_calculation](cell &cell) {
+    for (auto [side, b] : std::views::enumerate(cell.boundary)) {
+      auto o = orientation(side);
+      switch (b) {
+        case BoundaryCondition::outflow:
+          std::ranges::for_each(cell.particles, [&lc, &o](auto &e) { outflow<I>(lc, e, o); });
+          break;
+        case BoundaryCondition::reflecting:
+          std::ranges::for_each(cell.linear(), [&lc, &o, &force_calculation](auto &p) { reflecting<I>(lc, p, o, force_calculation); });
+          break;
+        case BoundaryCondition::none: break;
+      }
+    }
+  });
+}
+
+/**
+ * function to apply outflow boundary condition
+ * */
+template<index::Index I>
+static void outflow(linked_cell<I> &lc, arena<Particle>::entry &entry, orientation o) {
+
+  auto &p = entry.data;
+  const auto [pos_x, pos_y, pos_z] = p.position;
+  const auto [bound_x, bound_y, bound_z] = lc.index.boundary;
+
+  bool condition = false;
+  switch (o) {
+    case orientation::front:
+      condition |= pos_z > bound_z;
+      break;
+    case orientation::back:
+      condition |= pos_z < 0;
+      break;
+    case orientation::left:
+      condition |= pos_x < 0;
+      break;
+    case orientation::right:
+      condition |= pos_x > bound_x;
+      break;
+    case orientation::bottom:
+      condition |= pos_y < 0;
+      break;
+    case orientation::top:
+      condition |= pos_y > bound_y;
+      break;
+  }
+  entry.active &= !condition;
+}
+
+/**
+ * function to apply reflecting boundary condition
+ * */
+
+template<index::Index I>
+static void reflecting(linked_cell<I> &lc, auto &p, orientation o, auto fc) {
+  auto [pos_x, pos_y, pos_z] = p.position;
+  auto [bound_x, bound_y, bound_z] = lc.index.boundary;
+  auto diff = lc.index.boundary - p.position;
+  auto [diff_x, diff_y, diff_z] = diff;
+
+  const double sigma = lc.sigma;
+  const double distance = std::pow(2, 1 / 6.0) * sigma;

```

Comment:
If I am not mistaken, this possibly expensive expression is calculated for every (boundary) particle.